### PR TITLE
[SRVKS-482] Upgrade Knative Serving to 0.13.1.

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title, this contains a few vital bug fixes for us as well.

### CODE REVIEW

**Comment by markusthoemmes** [comment]
/retest

These things are very frustrating :/. These are likely proper bugs but I haven't found their root cause just yet.

**Comment by markusthoemmes** [comment]
/retest

infra issues.

**Comment by markusthoemmes** [comment]
/retest

Horrible infra issues.

**Comment by markusthoemmes** [comment]
FWIW: The root cause above has been resolved. Upstream updated the k8schain stuff even further and we needed to disable all chains now for it to work properly. That's fine on Openshift though.

File: test/lib.bash [hunk 1]
```diff
@@ -265,7 +265,7 @@ function run_knative_serving_operator_tests {
   logger.info "Checkout the code ${fork}/serving-operator @ ${version}"
   target="${GOPATH}/src/knative.dev/serving-operator"
   mkdir -p "$target"
-  git clone --branch "${version}" --depth 1 \
+  git clone --branch "openshift-${version}" --depth 1 \
```
Comment (mgencur) [inline test/lib.bash]: Hmm, I don't see as branch starting with `openshift-` in that repo.

Comment (mgencur) [reply test/lib.bash]: Oh. Sorry, this will take a tag as well.

**Comment by mgencur** [comment]
/lgtm
The serving-operator image is still registry.svc.ci.openshift.org/openshift/knative-v0.13.0:knative-serving-operator but I think this is expected because the CI publishes the snapshot of the image under knative-v0.13.0 repo

**Comment by markusthoemmes** [comment]
The tests all passed but we had a hang when deleting the KnativeServing.

I'll merge this regardless and will work on fixing that flake, we are finally able to see why it happened :).

**Pull request merged**